### ISSUE TITLE
AccessToken missing from LoginResponse

### ISSUE THREAD

**Issue by piccombs** [TheAxeDude/ZiniGo#2]
Nifty bit of code that I want to see work, but unless I'm missing something, they've changed the login response since you've last updated. The nature of API's I guess :/

I've been tinkering with your source and in the login response I'm seeing a data structure like this:

```
{"data":{"user":{"allow_adult_content":true,"default_currency_code":"USD","default_locale_code":"en_US","default_newsstand_id":101,"country_code":"US","first_name":null,"email":"example@email.com","user_id_string":"2357563771234567891","user_external_id":"id_example@email.com1234567891"}},"status":true}
```

Subsequent requests throw 401 errors because the access_token is missing in the json response of the login request. Not sure where to go from there.

**Comment by TheAxeDude** [comment]
Thanks for letting me know! I've taken a quick look and it definitely appears the response (and a few other things) have changed.

I will work through these and push a new version soon.

**Comment by TheAxeDude** [comment]
Hi,

Please can you try the version on the feature/testing-api-changes branch? Windows executable is here: https://github.com/TheAxeDude/ZiniGo/tree/feature/testing-api-changes/built 

If you need Linux, please let me know, I dont have access to a linux env at the moment to test it, so didnt push the artifact.

**Comment by Vinnie20201** [comment]
I have a feeling I may doing something wrong because I keep getting "executable file not found in %PATH%". Any idea what I could do? I keep pasting where to find Chrome.exe in filed explorer in -c.

**Comment by TheAxeDude** [comment]
Can you paste the whole command you're running? Feel free to remove any identifying info (usernames, passwords etc)

**Comment by Vinnie20201** [comment]
-u=Username -p=Password [-c="Program Files (x86)Google\Chrome\Application\Chrome.exe"]
This gives me the error  "cmd.Run() failed with exec: "google-chrome": executable file not found in %PATH%"

When I remove the brackets like so,
-u=Username -p=Password -c="Program Files (x86)Google\Chrome\Application\Chrome.exe"
I get the error  "cmd.Run() failed with exec: "Program Files (x86)Google\\Chrome\\Application\\Chrome.exe": file does not exist"

Sorry, I am not use to the cmd prompt.

**Comment by TheAxeDude** [comment]
That's great news! I think you just need to drop template.html (https://github.com/TheAxeDude/ZiniGo/blob/master/template.html) in the same folder as the executable.

**Comment by Vinnie20201** [comment]
I dropped the template.html file in z file that also contains the Zinigo_Windows_x64 application and it provides the same results (blank pdf with a page count of 1/1)

**Comment by TheAxeDude** [comment]
Thats strange, can you paste the logs from the app?

**Comment by Vinnie20201** [comment]
The command prompt shows me this 

C:\Users\user>C:\Users\user\issue\zinigo\Zinigo_Windows_x64.exe -u=Username -p=Password -c="C:\Program Files (x86)\Google\Chrome\Application\chrome.exe"
Starting the application...
Found 10 issues in library.
Loading HTML template
Source  https://cdn2.audiencemedia.com/1966e694bad90686516f99cdf432800fdca39290_1010/2d2180cfb28cbf7f08d3542e0aa82a350bdc04cb56916a316048f607b7639092/de4f2eaebe9aa7a38475145bc6b25dfec375431c30eca3aac051c04d0cee2665.svg
ID:  0
[C:\Program Files (x86)\Google\Chrome\Application\chrome.exe --headless --disable-gpu --print-to-pdf=C:\Users\gianc/issue/464549_0.pdf C:\Users\gianc/issue/464549_0.html]
Source  https://cdn2.audiencemedia.com/1966e694bad90686516f99cdf432800fdca39290_1010/2d2180cfb28cbf7f08d3542e0aa82a350bdc04cb56916a316048f607b7639092/1b20c7de0a9e911a7ac2334ebb37ee0ebc67d4cf0c65cbc34d1abe14093e6392.svg
ID:  1
[C:\Program Files (x86)\Google\Chrome\Application\chrome.exe --headless --disable-gpu --print-to-pdf=C:\Users\gianc/issue/464549_1.pdf C:\Users\gianc/issue/464549_1.html]
Source  https://cdn2.audiencemedia.com/1966e694bad90686516f99cdf432800fdca39290_1010/2d2180cfb28cbf7f08d3542e0aa82a350bdc04cb56916a316048f607b7639092/76479c2ad332c00a818c67ae7a5c03b0daf4d45534de17efb812a47c67b16bd3.svg

and it prints out pdfs that look like this, with different names.
![496938_69](https://user-images.githubusercontent.com/73567150/100127580-1cb91200-2e34-11eb-9a36-6269fd0c0cd5.png)

**Comment by TheAxeDude** [comment]
Thanks for that. I just tested on my side, and i replicated the issue when I removed the template.html. Can you confirm that template.html is in the same folder, and isnt empty? 

I will look at pushing a version with the default template built in.

Also, feel free to remove the majority of your comment - there might be some personal info you missed in the logs.

**Comment by Vinnie20201** [comment]
This is the folder, it is on my desktop and those are the contents of the template.html document. 

![Untitled](https://user-images.githubusercontent.com/73567150/100134395-02376680-2e3d-11eb-8141-4084e5d06654.png)

The issues folder is being pasted here
C:\Users\guser\issue

**Comment by TheAxeDude** [comment]
Thats so strange. I've just pushed a new version to the feature branch - it will use a default html template string. Can you try it out for me?

**Comment by Vinnie20201** [comment]
ok so the cmd shows this 

C:\Users\user>C:\Users\user\Desktop\zinigo\Zinigo_Windows_x64.exe -u=username -p=password -c="C:\Program Files (x86)\Google\Chrome\Application\chrome.exe"
Starting the application...
Found 10 issues in library.
Loading HTML template
template.html not found, or empty. using built in template. Consider changing this if your files are cropped.

Like the cmd said, no blanked pages, but all the pdfs were cropped.

**Comment by TheAxeDude** [comment]
That sounds better!

Can you make another folder (not on your desktop), like in c:/zinigo/ ? I suspect windows is doing something weird with the paths. Put the executable and the template html in there and run from there in the CMD window.

**Comment by Vinnie20201** [comment]
Zinigo is now located in C:\zinigo and the template HTML document is in there with it. I ran Zinigo exe in CMD and I got the same results.

**Comment by TheAxeDude** [comment]
Hmm. And does the log still say that it couldn't find the template html file? 

I'll need to add some extra logging so we can dig into it.

**Comment by Vinnie20201** [comment]
Yeah I am getting this still

Starting the application...
Found 10 issues in library.
Loading HTML template
template.html not found, or empty. using built in template. Consider changing this if your files are cropped.

Is there a way I am supposed to download the html file? I am finding it under ZiniGo/built/template.html then I click raw and save as.

**Comment by TheAxeDude** [comment]
100% correct. I'll add a new version with tracing so we can try figure out what's going on.

**Comment by Vinnie20201** [comment]
Ok keep me updated, thank you for all your help.

**Comment by TheAxeDude** [comment]
I have a new theory as to whats happening.

Can you please try the newest version from https://github.com/TheAxeDude/ZiniGo/tree/feature/testing-api-changes/built

I dont think it'll work, but it should print a message with "Resolved working directory to: xxx". Can you see if xxx matches where the program is stored?

**Comment by Vinnie20201** [comment]
You are correct. The CDM printed out this message.

Starting the application...
Found 10 issues in library.
Loading HTML template
template.html not found, or empty. using built in template. Consider changing this if your files are cropped.
Resolved working directory to: C:\Users\user

**Comment by TheAxeDude** [comment]
Okay cool, in your command window, type:

cd c:\Users\path\to\exe\folder

The 'cd' command means 'change directory'. Then please retry? It should then pickup the template.html....

Once thats done, you'll need to tweak the template.html to work for the dimensions of the magazine you're trying to download.

**Comment by Vinnie20201** [comment]
I got this in the cmd this time around. 

C:\Users\user>cd c:\zinigo

c:\zinigo>Zinigo_Windows_x64.exe -u=Username -p=Password -c="C:\Program Files (x86)\Google\Chrome\Application\chrome.exe"
Starting the application...
Found 10 issues in library.
Loading HTML template
template.html not found, or empty. using built in template. Consider changing this if your files are cropped.
Resolved working directory to: c:\zinigo

template.html, which is a chrome HTML Document, is located in there as well as Zinigo_Windows_x64, which is the application. 

Hope this helps.
